client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -473,4 +492,61 @@ Map<Integer, Set<Long>> getPartitionToBlockIds() {
   public WriteBufferManager getBufferManager() {
     return bufferManager;
   }
+
+  private static ShuffleManagerClient createShuffleManagerClient(String host, int port)
+      throws IOException {
+    ClientType grpc = ClientType.GRPC;
+    // Host can be inferred from `spark.driver.bindAddress`, which would be set when SparkContext is
+    // constructed.
+    return ShuffleManagerClientFactory.getInstance().createShuffleManagerClient(grpc, host, port);
+  }
+
+  private RssException throwFetchFailedIfNecessary(Exception e) {
```

Comment:
```suggestion
  private void throwFetchFailedIfNecessary(Exception e) {
```

Reply:
You don't get my point. You can return void and throw the exception which you need.

Reply:
> You don't get my point. You can return void and throw the exception which you need.

Understood, but here the writing is written with reference to read failure, if you want to change should be changed at the same time, to keep the code uniform. Same as above, the next PR is solving this problem?

Reply:
OK, let change this in the next pr.